server.go

```diff
@@ -66,6 +66,8 @@ type ServeConfig struct {
 	// the gRPC health checking service. This is not optional since go-plugin
 	// relies on this to implement Ping().
 	GRPCServer func([]grpc.ServerOption) *grpc.Server
+
+	Logger hclog.Logger
```

Comment:
Include a comment for `Logger` (perhaps mention in there that this is optional).